libs/base/Data/Vect.idr

```diff
@@ -181,9 +180,9 @@ merge = mergeBy compare
 ||| reverse [1,2,3,4]
 ||| ```
 public export
-reverse : Vect len elem -> Vect len elem
+reverse : (1 xs : Vect len elem) -> Vect len elem
 reverse xs = go [] xs
```

Comment:
Worth noting that we can have a [proof-free implementation of `reverse`
using `foldl`](http://agda.github.io/agda-stdlib/Data.Vec.Base.html#7735).

Reply:
While `foldr (flip cons)` would be great, it's again the issue that `foldr` and its higher-order friends only talk about unrestricted functions, so we have to implement the fold manually.